state-chain/pallets/cf-ingress-egress/src/lib.rs

```diff
@@ -1005,10 +1025,10 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 			deposit_channel_details.deposit_channel,
 		);
 		DepositBalances::<T, I>::mutate(asset, |deposits| {
-			deposits.register_deposit(net_deposit_amount)
+			deposits.register_deposit(deposit_amount)
```

Comment:
note here this is now using the full amount. The full amount is fetched, and therefore the full amount will be used when we register as fetched.

Reply:
I'm not sure this is correct. 

We *do* need to fetch the full (gross) deposit amount, but not all of that is actually deposited in the vault, since some of it is used for fees. 

I can see how the use of this isn't super clear, but the idea of the deposit balances is to track how much is available for swap egresses / lp withdrawals.

If we 'take' some of the deposit and put it in the withheld fees, we can't also put it in the deposits.

```
transfer_amount = deposit_amount + fees
^ 					^				^- Fees withheld at the witnessing step
|					|- amount actually credited to the vault and used in the swap/lp deposit 
|_ The total amount transferred by the user
```

I'll take a closer look at and may change this back.

Reply:
Would it make more sense to tie these two concepts a bit more closely together in that case?

Reply:
Maybe. We could also improve the naming/documentation and distinguish more clearly between what the user sends vs what we actually count as the deposit.

But IMO this is out of scope for this PR, especially since deposit tracking is not used anywhere. I added it as a placeholder to iterate upon, because I wanted us to have *something*, but it needs improvement.